### ISSUE TITLE
🐛 useSemanticElements should exclude table

### ISSUE THREAD

**Issue by unional** [biomejs/biome#6698]
### Environment information

```block
CLI:
  Version:                      2.0.6
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.9.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.11.1

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Workspace:
  Open Documents:               0
```

### What happened?

```tsx
<div role="grid"/>
```

This is flagged and recommend to use `<table>`.
It physically won't work as `<table>` has specific styling rule constraint making this not possible.

### Expected result

This should not be flag as an error.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
It feels like a weak argument though, especially if we consider that styling has nothing to do with the rule (and maybe a11y in general).

Plus, we're just following the spec.

Wouldn't this be a case where it's fine to suppress the rule?

**Comment by unional** [comment]
`table` (and its friends like `columnheader`) are really some special cases.
Their styling are applied differently than others and there is no way to use them.

To suppress the rule, it would basically need to be applied to every such instances (`role="table | grid | treegrid | columnheader | row | rowgroup | rowheader | cell | ...`).

Maybe adding an option to the rule to exclude them by default?

Ideally, suppression should be used for specific exeception case, IMO.

**Comment by sterliakov** [comment]
For reference, the upstream `jsx-a11y/prefer-tag-over-role` also rejects `<div role="table">`. DOM-level accessibility is not concerned about ease or possibility of styling. I think this should stay as-is and be suppressed on per-instance basis. If you do not consider this rule important for your use case, you can simply disable it globally, and that's fine - not all linter rules should be applied to every project in wild.

**Comment by ematipico** [comment]
Thank you. I think in this case the rule should be suppressed and/or disabled.
